pymc/smc/smc.py

```diff
@@ -546,6 +544,36 @@ def sample_settings(self):
         return stats
 
 
+def systematic(weights):
+    """
+    Systematic resampling.
+
+    Parameters
+    ----------
+    weights :
+        The weights should be probabilities and the total sum should be 1.
+
+    Returns
+    -------
+    new_indices: array
+        A vector of indices in the interval 0, ..., len(normalized_weights)
+    """
+    lnw = len(weights)
+    arange = np.arange(lnw)
+    uniform = (np.random.rand(1) + arange) / lnw
```

Comment:
This should use a numpy generator for seeding, which is passed to the function (`self.rng` in the case of SMC kernels)